section_faiss_vector_index.go

```diff
@@ -564,32 +564,41 @@ func (vo *vectorIndexOpaque) process(field index.VectorField, fieldID uint16, do
 	dim := field.Dims()
 	metric := field.Similarity()
 
-	if vec != nil {
```

Comment:
I don't think we should drop this `if vec != nil` to avoid any out of bounds reference below.

Reply:
We don't need to perform nil vec checks here, as bleve will already take care of it.